.github/workflows/docs.yml

```
@@ -0,0 +1,52 @@
+name: Build and Deploy API Documentation
+
+on:
+  push:
+    tags:
+      # Match 1.0.0 but not 1.0.0-rc1 or other suffixes
+      - '[1-9]+.[0-9]+.[0-9]+'
+
+jobs:
+  generate_docs:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+
+    steps:
+      - name: Checkout the branch with the latest code
+        uses: actions/checkout@v3
```

Should we use the v4 here as well?
```suggestion
        uses: actions/checkout@v4
```